pkg/query/storeset.go

```diff
@@ -527,15 +527,15 @@ func (s *StoreSet) updateStoreStatus(store *storeRef, err error) {
 		status = *prev
 	}
 
-	status.LastError = &stringError{originalErr: err}
-
 	if err == nil {
 		status.LastCheck = time.Now()
 		mint, maxt := store.TimeRange()
 		status.LabelSets = store.LabelSets()
 		status.StoreType = store.StoreType()
 		status.MinTime = mint
 		status.MaxTime = maxt
+	} else {
+		status.LastError = &stringError{originalErr: err}
```

Comment:
Don't we need to set `status.LastError` to `nil` if there was no error on this update?

Reply:
a new `status` is created a few lines up in this method, it's `LastError` will already be nil.

Reply:
But if it already exists then a new one is _not_ created, right?

```go
	prev, ok := s.storeStatuses[store.addr]
	if ok {
		status = *prev
	}
```

Reply:
Ah sorry yeah, missed that, thanks.

Have created a new unit test to highlight that and pushed a fix.